documents/coredocument.go

```diff
@@ -448,6 +465,56 @@ func (cd *CoreDocument) createProofs(fromZKTree bool, docType string, dataLeaves
 	}, nil
 }
 
+//CreateFingerprint generates a fingerprint for a Core Document
+func (cd *CoreDocument) CreateFingerprint() ([]byte, error) {
+	f := newFingerprint()
+	if len(cd.Document.Roles) == 0 {
+		return []byte{}, nil
+	}
+	if len(cd.Document.TransitionRules) == 0 {
+		return []byte{}, nil
+	}
+	f.Fingerprint.Roles = cd.Document.Roles
+	f.Fingerprint.TransitionRules = cd.Document.TransitionRules
+	p, err := cd.generateFingerprintHash(f.Fingerprint)
```

Comment:
`generateFingerprintHash` doesn't need to be a method to core document but instead can be a pure function